geoips/geoips_utils.py

```diff
@@ -75,91 +73,35 @@ def load_all_yaml_plugins():
-            # Absolute path to the plugin
-            yaml_plugin["abspath"] = str(yaml_file)
-
-            if "interface" not in yaml_plugin:
-                raise PluginError(
-                    f"YAML file encountered without 'interface' property: {yaml_file}"
-                )
-            if yaml_plugin["interface"] not in plugins:
-                plugins[yaml_plugin["interface"]] = [yaml_plugin]
-            else:
-                plugins[yaml_plugin["interface"]].append(yaml_plugin)
-    return plugins
-
-
-def find_config(subpackage_name, config_basename, txt_suffix=".yaml"):
-    """Find matching config file within GEOIPS packages.
-
-    Given 'subpackage_name', 'config_basename', and txt_suffix, find matching
-    text file within GEOIPS packages.
-
-    Parameters
-    ----------
-    subpackage_name : str
-        subdirectory under GEOIPS package to look for text file
-        ie text_fname = geoips/<subpackage_name>/<config_basename><txt_suffix>
-    config_basename : str
-        text basename to look for,
-        ie text_fname = geoips/<subpackage_name>/<config_basename><txt_suffix>
-    txt_suffix : str
-        suffix to look for on config file, defaults to ".yaml"
-        ie text_fname = geoips/<subpackage_name>/<config_basename><txt_suffix>
-
-    Returns
-    -------
-    text_fname : str
-        Full path to text filename
-    """
-    text_fname = None
-
-    for package_name in gpaths["GEOIPS_PACKAGES"]:
-        fname = os.path.join(
-            os.getenv("GEOIPS_PACKAGES_DIR"),
-            package_name,
-            subpackage_name,
-            config_basename + txt_suffix,
-        )
-        # LOG.info('Trying %s', fname)
-        if os.path.exists(fname):
-            LOG.info("FOUND %s", fname)
-            text_fname = fname
-        fname = os.path.join(
-            os.getenv("GEOIPS_PACKAGES_DIR"),
-            package_name,
-            package_name,
-            subpackage_name,
-            config_basename + txt_suffix,
+    reg_plug_path = resources.files("geoips") / "registered_plugins.yaml"
+    if not os.path.exists(reg_plug_path):
+        raise PluginRegistryError(
+            f"Plugin registry {reg_plug_path} did not exist, "
+            "please run 'create_plugin_registries'"
         )
-        # LOG.info('Trying %s', fname)
-        if os.path.exists(fname):
-            LOG.info("FOUND %s", fname)
-            text_fname = fname
-    return text_fname
+    plugin_packages = get_entry_point_group("geoips.plugin_packages")
+    yaml_plugins = {}
+    for pkg in plugin_packages:
+        pkg_plug_path = resources.files(pkg.value) / "registered_plugins.yaml"
+        # This will include all plugins, including schemas, yaml_based,
+        # and module_based plugins.
+        registered_plugins = yaml.safe_load(open(pkg_plug_path, "r"))
+        # Only pull the "yaml_based" plugins here.
+        try:
+            for interface in registered_plugins["yaml_based"]:
```

Comment:
This is where I resolved https://github.com/NRLMMD-GEOIPS/geoips/pull/240#discussion_r1367833487

Changed the plugin registries so they include yaml_based, module_based in the registry themselves (since we have that information in advance when we're creating it)